backend/src/ml_space_lambda/authorizer/lambda_function.py

```diff
@@ -324,6 +326,15 @@ def lambda_handler(event, context):
                                 policy_statement["Effect"] = "Allow"
                     elif target_type == DatasetType.PRIVATE and username == target_scope:
                         policy_statement["Effect"] = "Allow"
+                    elif target_type == DatasetType.GROUP:
+                        if Permission.ADMIN in user.permissions:
+                            policy_statement["Effect"] = "Allow"
```

Comment:
Just seeking some input…

We're doing a check like this…or at least `Permission.ADMIN in user.permissions` 10+ times in this file. Is there a scenario where we don't want the Admin to have access to something? We could clean up this file a bit by just putting a check for that at the beginning and not doing this check everywhere.

Reply:
What did you have in mind? Finding all of the places we do this check then condensing it into one conditional further up in the file? Like basically early on in this auth logic we'd first check if the user is an admin, then check if the requested resource is in some list of request resources that admins always have access to?

One argument in favor of the status quo is that it's very easy to follow. The logic is split up by request method and path params and I can easily find all of the dataset logic in one area, all of the job logic in one area, etc.

Reply:
What don't admins have access too? I feel like at the top of the file we can just say if you're an admin, allow it. Just being allowed to access the things doesn't change what we show admins in the UI. But I think they _should_ have access all resources right?

Reply:
I thought private datasets, but if I'm wrong and admins can touch literally everything then that sounds like a good idea